packages/client/src/utils/ShoppingCartContext/ShoppingCartContext.jsx

```diff
@@ -0,0 +1,64 @@
+import React, {
+  useState,
+  useContext,
+  useMemo,
+  useCallback,
+  createContext,
+} from 'react';
+import PropTypes from 'prop-types';
+import { useUserContext } from '../../userContext';
+
+const ShoppingCartContext = createContext({});
+
+export const useShoppingCart = () => {
+  return useContext(ShoppingCartContext);
+};
+
+const ShoppingCartProvider = ({ children }) => {
+  const { user } = useUserContext();
+
+  const [pendingItems, setPendingItems] = useState(() => {
+    const items = localStorage.getItem(user?.uid);
+    const initialValue = JSON.parse(items);
+    return initialValue || [];
+  });
+
+  const deletePendingItem = useCallback(
+    (itemId) => {
+      const filteredItems = pendingItems.filter(({ id }) => id !== itemId);
+      localStorage.setItem(user?.uid, JSON.stringify(filteredItems));
+      setPendingItems(filteredItems);
+    },
+    [pendingItems, user?.uid],
+  );
+
+  const addPendingItem = useCallback(
+    (test) => {
+      const updatedPendingItems = { ...pendingItems, ...test };
```

Comment:
`pendingItems` is a list, but here you spread it into an object, that's probably a mistake.
Further, `test` is probably and object representing a product so I don't think you want to spread that apart.